### PR TITLE
feat(components) create <TextField/>

### PR DESCRIPTION
**Pull request by AndriiTsarenko**

### CODE REVIEW

File: src/components/TextField/docs/index.story.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  component: TextField,
+  componentPath: '../TextField.tsx',
+
+  componentProps: (setState, getState) => ({
+    'data-hook': 'storybook-TextField',
+    value: '',
+    onChange: ({ target: { value } }) => setState({ value }),
+  }),
+
+  examples: <Examples />,
```
Comment (jonathanadler) [inline src/components/TextField/docs/index.story.tsx]: Let's try to use the same storybook format as the other components (Textarea) for instance.
Using sections and our custom `settingsPanel` method

File: src/components/TextField/TextField.driver.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      return isSuccessValue();
+    },
+    isError() {
+      return isErrorValue();
+    },
+    getErrorMessage() {
+      return getErrorMessage();
+    },
+    hasEmptyState() {
+      return stylableDOMUtil.hasStyleState(element, 'empty');
```
[the remark this answers is not in the document]
Comment (AndriiTsarenko) [reply src/components/TextField/TextField.driver.ts]: Done

File: src/components/TextField/TextField.st.css [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    -st-named: Input;
+}
+
+:import {
+    -st-from: "../../common/palette.st.css";
+    -st-named: errorColor, successColor;
+}
+
+:vars {
+    MainBorderColor: applyStyleMethod(color, color-5);
```
[the remark this answers is not in the document]
Comment (AndriiTsarenko) [reply src/components/TextField/TextField.st.css]: Fixed

File: src/components/TextField/TextField.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+describe('TextField', () => {
+  const createDriver = createDriverFactory(textFieldDriverFactory);
+
+  it('should render', () => {
+    const value = 'hello!';
+    const driver = createDriver(<TextField value={value} />);
+    expect(driver.getValue()).toEqual(value);
+  });
+
```
[the remark this answers is not in the document]
Comment (AndriiTsarenko) [reply src/components/TextField/TextField.spec.tsx]: I've added a few tests more.

File: src/components/TextField/docs/index.story.tsx (hunk 1 above)
Comment (AndriiTsarenko) [reply src/components/TextField/docs/index.story.tsx]: Done

File: src/components/TextField/TextField.driver.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import { inputDriverFactory as coreDriver } from 'wix-ui-core/drivers/vanilla';
+import { StylableDOMUtil } from '@stylable/dom-test-kit';
+import style from './TextField.st.css';
```
Comment (jonathanadler) [inline src/components/TextField/TextField.driver.ts]: `StylableDOMUtil` and `style` can be removed, they're not used

File: src/components/TextField/TextField.protractor.driver.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+export interface TextFieldDriver extends InputDriver {
+  hover();
+}
+
+export const textFieldDriverFactory = (component): TextFieldDriver => {
+  return {
+    ...inputDriverFactory(component),
+    async hover() {
+      const input = component.$('input');
+      console.log(component.browser);
```
Comment (jonathanadler) [inline src/components/TextField/TextField.protractor.driver.ts]: Just need to remove logs and comments

File: src/components/TextField/TextField.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    [ERROR]: error,
+    [EMPTY]: !props.value,
+  };
+  const test = (props as any).test;
+  console.info(test);
+  return (
+    <CoreInput
+      {...dataObject}
+      {...style('root', {
+        test, //needed for visual test. hides text caret removes transitions
```
Comment (jonathanadler) [inline src/components/TextField/TextField.tsx]: Can you explain the `test` prop?
If it's only for hiding the caret, I think we can configure applitools for it.
We prefer to not add production code that's used only for testing.

File: src/components/TextField/TextField.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+  }
+  const dataObject = {
+    [ERROR_MESSAGE]: errorMessage,
+    [THEME]: theme,
+    [SUCCESS]: success,
+    [ERROR]: error,
+    [EMPTY]: !props.value,
+  };
+  const test = (props as any).test;
+  console.info(test);
```
Comment (jonathanadler) [inline src/components/TextField/TextField.tsx]: `console` should be removed

File: src/components/TextField/TextField.tsx (hunk 7 above)
Comment (AndriiTsarenko) [reply src/components/TextField/TextField.tsx]: It also removes the `transition` from `border`. We can configure the Aplitools to handle both text caret and transitions?

File: src/components/TextField/TextField.protractor.driver.ts (hunk 6 above)
Comment (AndriiTsarenko) [reply src/components/TextField/TextField.protractor.driver.ts]: It's done.

File: src/components/TextField/TextField.tsx (hunk 7 above)
Comment (jonathanadler) [reply src/components/TextField/TextField.tsx]: I think we can configure the test to ignore the caret.
Regarding transitions, we can add a short timeout before the snapshot for this.
Nevertheless, this code should be removed.

Comment (AndriiTsarenko) [reply src/components/TextField/TextField.tsx]: removed.

File: src/components/TextField/TextField.st.css [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+.root:focus:error {
+    border-color: value(errorBorderColorFocus);
+}
+
+.root:theme(line) {
+    border-top: none;
+    border-left: none;
+    border-right: none;
+    padding: 8px 0;
+    background-color: transparent;
```
Comment (jonathanadler) [inline src/components/TextField/TextField.st.css]: Designer QA note:
the `line` theme should have a background-color that can be overridable, like this:
```background-color: color(fallback(MainBackgroundColor, transparent))``` 
Note the `MainBackgroundColor` with the big `M`

Comment (AndriiTsarenko) [reply src/components/TextField/TextField.st.css]: Done.

**Pull request merged**